ui/main.qml

```diff
@@ -367,12 +379,104 @@ StatusWindow {
         }
     }
 
-    OnboardingLayout {
-        id: startupOnboarding
-        objectName: "startupOnboardingLayout"
+    Component {
+        id: splashScreenV2
+        DidYouKnowSplashScreen {
+            readonly property string pageClassName: "Splash"
+            property bool runningProgressAnimation
+            NumberAnimation on progress {
+                from: 0.0
+                to: 1
+                // TODO find a way to unhardcode this
+                // Though there is no easy way to do it, because we do not get progress signals
+                duration: 3000
+                running: runningProgressAnimation
+            }
+        }
+    }
+
+    Loader {
+        id: onboardingStoreLoader
+        active: applicationWindow.onboardingV2Enabled
+
+        sourceComponent: OnboardingStore {
+            onAppLoaded: moveToAppMain()
+        }
+    }
+
+    Loader {
+        id: startupOnboardingLoader
         anchors.fill: parent
+        sourceComponent: {
+            if (applicationWindow.onboardingV2Enabled) {
+                // TODO select a new component when we have the new login screens (for old users)
+                return onboardingV2
+            }
+            return onboardingV1
+        }
+    }
+
+    Component {
+        id: onboardingV1
+
+        OnboardingLayout {
+            objectName: "startupOnboardingLayout"
+            anchors.fill: parent
+
+            utilsStore: applicationWindow.utilsStore
+        }
+    }
+
+    Component {
+        id: onboardingV2
+
+        Onboarding2.OnboardingLayout {
+            objectName: "startupOnboardingLayout"
+            anchors.fill: parent
+
+            networkChecksEnabled: true
+            biometricsAvailable: Qt.platform.os === Constants.mac
+
+            onboardingStore: onboardingStoreLoader.item
```

Comment:
I'd just put the store directly here; no need for it to be a separate Loader imo